Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionTest.cs

```diff
@@ -14,13 +14,491 @@
+      _domainBaseTypeDefinition.SetupForReadOnlyUsage();
+      _domainBaseTypeDefinition.SetReadOnly();
+
+      var result1 = _domainBaseTypeDefinition.GetPropertyDefinitions();
+      var result2 = _domainBaseTypeDefinition.GetPropertyDefinitions();
+
+      Assert.That(result1, Is.Not.Null);
+      Assert.That(result1, Is.SameAs(result2));
+    }
+
+    [Test]
+    public void GetPropertyDefinition ()
+    {
+      var propertyInformation = PropertyDefinitionObjectMother.CreateForFakePropertyInfo(_domainBaseTypeDefinition);
+      _createPropertyDefinitionCollectionMock.Setup(_ => _()).Returns(new PropertyDefinitionCollection(new []{propertyInformation}, true));
+
+      Assert.That(
+          _domainBaseTypeDefinition.GetPropertyDefinition("Test"),
+          Is.SameAs(propertyInformation));
+    }
+
+    [Test]
+    public void GetPropertyDefinition_EmptyPropertyName_Throws ()
+    {
+      Assert.That(
+          () => _domainBaseTypeDefinition.GetPropertyDefinition(string.Empty),
+          Throws.ArgumentException
+              .With.ArgumentExceptionMessageEqualTo("Parameter 'propertyName' cannot be empty.", "propertyName"));
+    }
+
+    [Test]
+    public void GetPropertyDefinition_NonExistingProperty_ReturnsNull ()
+    {
+      _createPropertyDefinitionCollectionMock.Setup(_ => _()).Returns(new PropertyDefinitionCollection());
+
+      Assert.That(_domainBaseTypeDefinition.GetPropertyDefinition("dummy"), Is.Null);
+    }
+
+    [Test]
+    public void GetMandatoryPropertyDefinition_NonExistingProperty_Throws ()
+    {
+      _createPropertyDefinitionCollectionMock.Setup(_ => _()).Returns(new PropertyDefinitionCollection());
+
+      Assert.That(
+          () => _domainBaseTypeDefinition.GetMandatoryPropertyDefinition("dummy"),
+          Throws.TypeOf<MappingException>()
+              .With.Message.EqualTo(
+                  "Type 'Remotion.Data.DomainObjects.UnitTests.Mapping.TypeDefinitionTest+IDomainBase' "
+                  + "does not contain the property 'dummy'."));
+    }
+
+    [Test]
+    public void SetPropertyDefinitions ()
+    {
+      var propertyDefinition = PropertyDefinitionObjectMother.CreateForFakePropertyInfo(_domainBaseTypeDefinition);
+
+      _checkPropertyDefinitionsMock
+          .Setup(_ => _(new[] { propertyDefinition }))
+          .Callback(
+              () => { Assert.That(() => _domainBaseTypeDefinition.MyPropertyDefinitions, Throws.Exception); }
+          );
+
+      _domainBaseTypeDefinition.SetPropertyDefinitions(new PropertyDefinitionCollection(new[] { propertyDefinition }, false));
+
+      Assert.That(_domainBaseTypeDefinition.MyPropertyDefinitions.Count, Is.EqualTo(1));
+      Assert.That(_domainBaseTypeDefinition.MyPropertyDefinitions[0], Is.SameAs(propertyDefinition));
+      Assert.That(_domainBaseTypeDefinition.MyPropertyDefinitions.IsReadOnly, Is.True);
+
+      _checkPropertyDefinitionsMock.Verify(_ => _(new[] { propertyDefinition }));
+    }
```

Comment:
Based on symmetry with other tests, we'd use Verifiable() + Verify() instead of AAA syntax. Is there a particular benefit to using AAA?